src/kaocha/plugin.clj

```diff
@@ -16,22 +16,48 @@
     (catch java.io.FileNotFoundException e
       false)))
 
+(defn- try-require-info [n]
+  [(try-require n) n])
+
 (defn try-load-third-party-lib [plugin-name]
-  (if (qualified-keyword? plugin-name)
-    (when-not (try-require (symbol (str (namespace plugin-name) "." (name plugin-name))))
-      (try-require (symbol (namespace plugin-name))))
-    (try-require (symbol (name plugin-name)))))
+  (let [full-name (symbol (str (namespace plugin-name) "." (name plugin-name)))]
+    (if (qualified-keyword? plugin-name)
+      (if-let [full-result (try-require full-name)]
+        [[full-result full-name]]
+        [[false full-name] (try-require-info (symbol (namespace plugin-name))) ])
+      [(try-require-info (symbol (name plugin-name)))])))
 
 (defmulti -register "Add your plugin to the stack"
   (fn [name plugins] name))
 
 (defmethod -register :default [name plugins]
-  (output/error "Couldn't load plugin " name)
   (throw+ {:kaocha/early-exit 254} nil (str "Couldn't load plugin " name)))
 
 (defn register [plugin-name plugin-stack]
-  (try-load-third-party-lib plugin-name)
-  (-register plugin-name plugin-stack))
+  (let [ns-result (try-load-third-party-lib plugin-name)
+        successful-ns (map second (filter first ns-result))
+        failed-ns (map second (filter (complement first) ns-result))
+        plugin-result (try+ (-register plugin-name plugin-stack)
+                            (catch [:kaocha/early-exit 254] _ false)) ] 
+    (cond
+      ;Namespaces succeeded, but plugin itself failed.
+      (and (not plugin-result)
+           (every? true? (map first ns-result))) (output/error-and-throw {:kaocha/early-exit 254} nil
+                                                  (format "Couldn't load plugin %s, but loaded %s." 
+                                                          plugin-name (first successful-ns))) 
```

Comment:
I think we can make this a little more clear and explicit.

`Failed loading plugin %s. The plugin was not defined after loading the %s namespace.`

Reply:
Yeah, I think that's better. I also added more explanation while I was at it.

It now reads `Couldn't load plugin %s. The plugin was not defined after loading namespace %s. Is the file missing a defplugin?`